redis/tests/test_basic.rs

```diff
@@ -667,13 +670,50 @@ fn test_pubsub() {
     assert_eq!(con.publish("foo", 23), Ok(1));
 
     thread.join().expect("Something went wrong");
+    if ctx.protocol == ProtocolVersion::RESP3 {
```

Comment:
I'd prefer if this was in a test that didn't use a `PubSub` connection, but rather a test that demonstrated how this works on a regular connection.

Reply:
Hi, I think this test should stay as is and there is already a test called `test_push_manager`. I don't think currently there will be usage for PushManager other than PubSub until Client Side Caching.

Reply:
part of the point of push manager is that it can be used with a regular connection, not only with a `PubSub` connection. Notice the distinction - I'm not talking about using the push manager to get pub-sub messages, but to use it with a different struct - `redis::Connection` instead of `redis::PubSub`. 
ATM there's no example on how to do that, not in tests and not in documentation. `test_push_manager` demonstrates how to use the push manager for client tracking, but not for pub-sub, which IMO is a more common usecase.

Reply:
I have added simple test, but in future I want to add examples to examples folder

Reply:
The library's documentation is also a great place for examples :)